sys/include/net/gcoap.h

```diff
@@ -1045,6 +1046,320 @@ ssize_t gcoap_encode_link(const coap_resource_t *resource, char *buf,
 sock_dtls_t *gcoap_get_sock_dtls(void);
 #endif
 
+/**
+ * @defgroup net_gcoap_nanocoap gCoAP packet abstraction layer
+ * @ingroup net_gcoap
+ *
+ * gCoAP utilizes @ref net_nanocoap to parse and abstract CoAP packets. To separate those to modules
+ * cleanly, this module provides wrappers around nanocoap's functionality.
+ * @{
+ */
+/**
+ * @name    Buffer manipulation
+ * @{
+ */
+static inline uint8_t *gcoap_pkt_get_buf(coap_pkt_t *pdu)
```

Comment:
```suggestion
static inline void *gcoap_pkt_get_buf(coap_pkt_t *pdu)
```
should be more flexible